test/serializer/pure-functions/AbstractObject.js

```diff
@@ -0,0 +1,19 @@
+let c = global.__abstract ? __abstract('boolean', '(true)') : true;
+
+function test(fn) {
+  let knownObj = {x:1};
+  let knownObj2 = {y:3};
+  let conditionalObj = c ? knownObj : knownObj2;
+  fn(conditionalObj); // This must havoc both known objects
```

Comment:
I guess this is the case where we have concrete objects in the value domain, and the second half of visitAbstractValue kicks in.